### PR TITLE
Add Entry type checking utilities.

### PR DESCRIPTION
**Pull request by ostromart**
These functions have been in use throughout the ocstructs and validation libraries but really belong with goyang Entry type.

### CODE REVIEW

File: pkg/yang/entry.go [hunk 1]
```diff
@@ -110,6 +110,13 @@ type RPCEntry struct {
 	Output *Entry
 }
 
+// A ListAttr is associated with an Entry that represents a List node
+type ListAttr struct {
+	MinElements *Value // leaf-list or list MUST have at least min-elements
+	MaxElements *Value // leaf-list or list has at most max-elements
+	OrderedBy   *Value // order of entries determined by "system" or "user"
+}
+
```
Comment (pborman) [inline pkg/yang/entry.go]: I assume there will be another pull request that actually uses this type?

Comment (ostromart) [inline pkg/yang/entry.go]: You mean ListAttr or the below functions? ListAttr is just moved to get it out of the way to Modules(). The functions below are used in ocstructs which is slated for OS release soon - that's where the change will be made to use the functions here instead.

Comment (pborman) [inline pkg/yang/entry.go]: Ahh, yes, I missed the "deleted lines" in the left column (am too used to seeing moved lines marked as moved).
